### ISSUE TITLE
nullinvalid_token

### ISSUE THREAD

**Issue by arcreative** [ajvwhite/homebridge-smartthings-routine-triggers#3]
This is occurring on both my and your plugins now... not sure if you have a resolution, but looks like the API may have changed again?

**Comment by arcreative** [comment]
Also not sure why the config won't run from the simulator--it's been a while since I've messed with this, but would be much easier to cut and paste from a computer instead of a phone!  Any luck with that?

**Comment by jrdpizzo** [comment]
Same here. Smart app provides nullinvalid_token instead of the Config.

**Comment by ajvwhite** [comment]
Hi @jrdpizzo & @arcreative - I just created a fresh version of the app on the online IDE and didn't have any issues, it gave me the config as expected.

Which Graph API are you hitting the EU / US one? It may be that something may have changed on the US one, as I am on the EU one, I can not check as it forces me into the EU region.

@arcreative I actually just use the copy / paste handoff baked into iOS or alternatively an SSH client on my device, so actually not attempted to use the simulator.

This is still working at least on the EU side, but happy to try help you debug further.

**Comment by arcreative** [comment]
@ajvwhite I'm on the US API, looks like that might be the difference?

**Comment by raman325** [comment]
Ran into the same issue. It looks like the URL being called to generate the config was hardcoded to query the EU URL. PR #4 will fix the issue

**Comment by atrus05** [comment]
I am not convinced it is a EU vs US thing. I am on EU, and it says nullinvalid_token for me as well. I tried your PR @raman325 and it still throws an nullinvalid_token message when I try to get the config file.
Could anyone please let me paste the structure of the config (with redacted keys of course) so I can try this one out?

**Comment by raman325** [comment]
@atrus05 I don't know enough to troubleshoot your particular problem but the original code hardcoded a reference to the EU endpoint for SmartThings which would be a problem for non EU users either way. I'm based out of NA and the changes in the PR resolved my issues.

This may be a dumb question but how did you create the SmartApp in SmartThings? Did you copy and paste the raw code from my PR or did you use the github integration? The github integration will not include my changes yet since the PR has not been accepted, and I made the PR against @ajvwhite 's develop branch so he will have to merge to master after he accepts the PR (if he does)

Also: did you enable OAuth for the SmartApp when you set it up?

**Comment by ajvwhite** [comment]
> Ran into the same issue. It looks like the URL being called to generate the config was hardcoded to query the EU URL. PR #4 will fix the issue

@raman325 Yeah, this is mentioned on line 17 of the [README.md](https://github.com/ajvwhite/homebridge-smartthings-routine-triggers/blob/master/README.md) - mentioning if you were outside of the EU to replace the base API.

IIRC at the time this was written there wasn't a reliable way to determine this, at least not one that seemed to work... but it's been a while, so could have changed!

I'll take a look at the PR, and thanks for the contribution, need to look at the other PR too, so will try do both at the same time!

@atrus05 The error typically occurs if you have not activated OAuth for the app as @raman325 stated.

**Comment by raman325** [comment]
You know, I read the README but completely missed that part, whoops! Anyway, the PR I submitted *should* allow you to remove that part of the instructions, but feel free to take your time, I already made the changes I needed to get my SmartApp working 😄
